src/schemas/Banner/index.ts

```diff
@@ -0,0 +1,98 @@
+import { gql } from 'apollo-server-express';
+import { makeExecutableSchema } from 'graphql-tools';
+import { createCheckers } from "ts-interface-checker";
+import logger from '../../utils/logger';
+import bannerTI from "./banner-ti";
+
+const { BannerTI } = createCheckers(bannerTI);
+
+type AllowedFields =
+  | 'dismissable' 
+  | 'id' 
+  | 'level'
+  | 'message'
+  | 'title'
+;
+
+interface Banner {
+  dismissable: boolean;
+  id: string;
+  level: 'error'|'info'|'warning';
+  message?: string;
+  title: string;
+}
+
+const allowedFields = [
+  'dismissable',
+  'id',
+  'level',
+  'message',
+  'title',
+];
```

Comment:
if you type this as `AllowedFields[]`, it'll help avoid mistakes in forgetting to add fields when this changes ;)

Reply:
actually, `AllowedFields` can just be `keyof Banner` too

Reply:
god definition i like that

Reply:
i updated `AllowedFields` to be `keyof Banner` and added a comment about the other issue

tl;dr we're doing runtime checks because there's no other data validation for the banners and we don't want the build to crash